cclib/method/ddec.py

```diff
@@ -631,67 +685,258 @@ def _calculate_w_and_u(self):
+                        elif 3 * abs(belowphi) < abs(abovephi):
+                            corbigPhi = roots.min() - 2.0 * belowphi * (
+                                roots.max() - roots.min()
+                            ) / (abovephi - belowphi)
+                        else:
+                            corbigPhi = (roots.max() + roots.min()) / 2.0
+                        # New candidates
+                        corphi = self._phiaii(self.rho_wavg[atomi], eta, corbigPhi, atomi)[0]
+                        self._candidates_bigPhi[atomi] = numpy.array(
+                            [
+                                lowerbigPhi,
+                                midbigPhi,
+                                upperbigPhi,
+                                roots.max(),
+                                roots.min(),
+                                corbigPhi,
+                            ],
+                            dtype=float,
+                        )
+                        self._candidates_phi[atomi] = numpy.array(
+                            [lowerphi, midphi, upperphi, abovephi, belowphi, corphi], dtype=float
+                        )
+
+                        # Update upperphi and lowerphi
+                        lower_ind = numpy.where(
+                            self._candidates_phi[atomi]
+                            == self._candidates_phi[atomi][self._candidates_phi[atomi] < 0].max()
+                        )[0][0]
+                        upper_ind = numpy.where(
+                            self._candidates_phi[atomi]
+                            == self._candidates_phi[atomi][self._candidates_phi[atomi] > 0].min()
+                        )[0][0]
+
+                        lowerphi = self._candidates_phi[atomi][lower_ind]
+                        upperphi = self._candidates_phi[atomi][upper_ind]
+
+                        if abs(lowerphi) < self.convergence_level:
+                            bigphiAII[atomi] = self._candidates_bigPhi[atomi][lower_ind]
+                            phiconverged = True
+                        elif abs(upperphi) < self.convergence_level:
+                            bigphiAII[atomi] = self._candidates_bigPhi[atomi][upper_ind]
+                            phiconverged = True
+                        else:
+                            # Fitting needs to continue in this case.
+                            lowerbigPhi = self._candidates_bigPhi[atomi][lower_ind]
+                            lowerphi = self._candidates_phi[atomi][lower_ind]
+                            upperbigPhi = self._candidates_bigPhi[atomi][upper_ind]
+                            upperphi = self._candidates_phi[atomi][upper_ind]
+
+            assert phiconverged, "Iterative conditioning failed to converge."
+
+            # Set final G_A value using chosen Phi
+            self._g[atomi] = self._phiaii(self.rho_wavg[atomi], eta, bigphiAII[atomi], atomi)[1]
+
+    def calculate_H(self):
```

Comment:
Is there any interpretation for this H quantity? It would be helpful to know what is hiding behind the symbol

Reply:
I've added a comment on both reshape_G and calculate_H for more clarity. Basically, both G an H are constraints set on the tail of the proatom density.